aten/src/ATen/native/transformers/hip/flash_attn/CMakeLists.txt

```diff
@@ -0,0 +1,54 @@
+include(CMakePrintHelpers)
+# validate user-specified fmha_fwd API list
+set(FMHA_FWD_KNOWN_APIS "fwd;fwd_splitkv;fwd_appendkv")
+set(FMHA_FWD_ENABLE_APIS "fwd" CACHE STRING
+    "semicolon-separated list of APIs to generate (${FMHA_FWD_KNOWN_APIS}) & link, or \"all\".")
+if(FMHA_FWD_ENABLE_APIS STREQUAL "all")
+  set(FMHA_FWD_ENABLE_APIS ${FMHA_FWD_KNOWN_APIS})
+endif()
+
+foreach(api ${FMHA_FWD_ENABLE_APIS})
+  if(NOT "${api}" IN_LIST FMHA_FWD_KNOWN_APIS)
+    message(FATAL_ERROR "${api} isn't a known api: ${FMHA_FWD_KNOWN_APIS}.")
+  endif()
+endforeach()
+
+# "fwd" is a must-have api for the fmha_fwd example, add it if not specified
+if(NOT "fwd" IN_LIST FMHA_FWD_ENABLE_APIS)
+  list(APPEND FMHA_FWD_ENABLE_APIS "fwd")
+endif()
+
+string(REPLACE ";" "," FMHA_FWD_APIS "${FMHA_FWD_ENABLE_APIS}")
+cmake_print_variables(FMHA_FWD_APIS)
+# generate a list of kernels, but not actually emit files at config sta
+execute_process(
+  COMMAND python3 ${CMAKE_CURRENT_LIST_DIR}/../../../../../../../third_party/composable_kernel/example/ck_tile/01_fmha/generate.py
+  --api fwd,fwd_splitkv,fwd_appendkv --list_blobs ${CMAKE_CURRENT_LIST_DIR}/fwd_blob_list.txt
+  RESULT_VARIABLE ret
+)
+if(ret AND NOT ret EQUAL 0)
+  message( FATAL_ERROR "CK Tile FMHA FAILED to generate a list of FWD kernels via Python.")
+endif()
+
+execute_process(
+  COMMAND python3 ${CMAKE_CURRENT_LIST_DIR}/../../../../../../../third_party/composable_kernel/example/ck_tile/01_fmha/generate.py
```

Comment:
codegen in our internal system is kind of tricky. Any chance we can pre-generate the instance files and check them in?

Reply:
technically we could, but that would require checking in over 1400+ files. Additionally, the kernels need to be regenerated every time CK is updated which would introduce some technical debt.

Reply:
I think it's fine. This is what we do for cuda i think? https://github.com/pytorch/pytorch/tree/main/aten/src/ATen/native/transformers/cuda/flash_attn/kernels. Though 1400 is a lot :)  I'd strongly prefer checking in the file (at least for internal adoption - for glob generated files, we'll need an exact list of the files; so we'll need to update our script every single time we update CK which is a lot of work; also the CK version we use internally is different from oss pytorch.

Reply:
I will check in the blob files. Are you using the latest `develop` branch for CK internally?

Reply:
cc @malfet

Reply:
also cc @r-barnes

Reply:
discussed internally. Let's check in the generated files for now. We can improve in the future. But just remember we need to add those lines at the beginning

// ==========================================
// THIS CODE IS AUTOGENERATED. DO NOT MODIFY.
// @generated
// ==========================================

Reply:
There are quite a bit more files than i originally quoted you. The previous number came from the first prototype. After updating CK the last time, looks like the associated generate.py script made even more than before. in any case, i committed them and added the header you suggested

Reply:
@malfet Please NOTE: this PR will add 3552 kernel files to the OSS source code.